path/__init__.py

```diff
@@ -231,10 +179,12 @@ class Path(text_type):
     .. seealso:: :mod:`os.path`
     """
 
-    def __init__(self, other=''):
+    def __new__(cls, other=''):
```

Comment:
I'm not sure what this change does, and it's not mentioned in the changelog. Please clarify why the prior implementation was inadequate.

Reply:
About this change, take a look here: 
[https://stackoverflow.com/questions/7255655/how-to-subclass-str-in-python](https://stackoverflow.com/questions/7255655/how-to-subclass-str-in-python).

In addition,  `pytlint` shows :

- **pylint**: 
```python
def __init__(self, other=''):
```
  - *strsub.py:6:4: W0231*:   __init__ method from base class 'str' is not called (super-init-not-called)

Reply:
So the pylint warning is warning that the implementation has overridden `__init__`, but not called `super().__init__`. Addressing that lint warning directly is probably preferable. I suggest:

```
super(Path, self).__init__()
```

More importantly, fixing this issue seems unrelated to the goal of the PR at hand. Let's separate those concerns into separate requests.